testing/kokoro/system_tests.sh

```diff
@@ -196,8 +196,8 @@ elif [[ -z "${TARGET_DIRS// }" ]]; then
   GO_TEST_MODULES="./go.mod"
   echo "Only running root tests"
 else
-  GO_TEST_TARGET="./..."
-  GO_TEST_MODULES="$GO_CHANGED_SUBMODULES"
+  GO_TEST_TARGET=$TARGET_DIRS
+  GO_TEST_MODULES="./go.mod"
```

Comment:
This seems like it would prevent tests running in modules.

Reply:
I _think_ (from running `go test ./storage/...`), this will run tests in different modules. I could be wrong? Or maybe it uses the wrong `go.mod`?

Reply:
Oh wait. I'm wrong. `storage` isn't a different module (it is for cloud.google.com/go).

```
go test ./run/image-processing/...
go: directory run/image-processing is outside main module
go: directory run/image-processing/imagemagick is outside main module
```

So, this needs to change.

Reply:
Updated this. I removed the indirection between defining the target/modules and running the test. That makes it a bit easier to run the necessary tests directly. To make sure we use consistent `go test` invocations (including XML translation), I put the `go test` invocation into a shared function.